cmake/OpenCVDetectCUDA.cmake

```diff
@@ -1,3 +1,50 @@
+if(ENABLE_CUDA_FIRST_CLASS_LANGUAGE)
```

Comment:
There are 375 changes in this file which size is 577 lines on 4.x branch (more that 50%).

To avoid mess and/or breaking of existed legacy code my proposal is creating a copy of this file, cleanup dead code, and specify code for the new CUDA support.